README.md

```diff
@@ -31,3 +31,35 @@ Once you've imported our library, make sure your API key is available in the HYP
 That's it! Open the Hyperdash mobile app and you should see your logs start to stream in!
 
 Don't have an API key? Run `hyperdash signup` to get one!
+
+### API key storage
+
+If you signed up through the CLI, then your API key is already installed in hyperdash.json file in the home directory of your user.
+
+You can alternatively override this API key with a hyperdash.json file in your local directory (so you can have different API keys for different projects) or with the HYPERDASH_API_KEY environment variable.
+
+Finally, the monitor function accepts an api_key_getter argument that if passed in will be called everytime we need to retrieve your API key. Example:
+
+```
+# test_script.py
+
+from hyperdash.sdk import monitor
+
+# This function can do anything you want, as long as it returns a Hyperdash API key as a string
+def get_api_key():
+  return "super_secret_api_key"
+
+@monitor("dogs vs. cats", api_key_getter=get_api_key)
```

Comment:
Since this is being distributed, do we want to think about the calling arguments here? Just thinking out loud. 

What Andrew said about tagging resonated. It would be super nice to know (from the mobile app) some broad information about my job. e.g.,
* Platform: Keras/Tensorflow
* Commit SHA: JQL90I3
* Data source: cats_vs_dogs_20161101.csv
* etc.

We could represent this like 

```python
hd_options = {
  "source": "Keras/Tensorflow",
  "commit": ...
}
# ...
@monitor("dogs vs. cats", hd_options)
```

Additionally, to support custom actions, do we want to use this as an entrypoint or have them self-registerable. 

```python
from hyperdash import action

@action("Shutdown Server")
def shutdown_server():
  os.system("some_command with args")
```